cmd/ttn-lw-cli/commands/end_device_templates.go

```diff
@@ -0,0 +1,436 @@
+		}),
+	}
+	endDeviceTemplatesAssignEUIsCommand = &cobra.Command{
+		Use:     "assign-euis [join-eui] [start-dev-eui] [flags]",
+		Aliases: []string{"euis"},
+		Short:   "Assign JoinEUI and DevEUIs to end device templates",
+		Long: `Assign JoinEUI and DevEUIs to end device templates
+
+This command takes an end device template from stdin.`,
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if inputDecoder == nil {
+				return nil
+			}
+
+			forwardDeprecatedDeviceFlags(cmd.Flags())
+			joinEUIHex, _ := cmd.Flags().GetString("join-eui")
+			startDevEUIHex, _ := cmd.Flags().GetString("start-dev-eui")
+			switch len(args) {
+			case 0:
+			case 1:
+				logger.Warn("only single EUI found in arguments, not considering arguments")
+			case 2:
+				joinEUIHex = args[0]
+				startDevEUIHex = args[1]
+			default:
+				logger.Warn("multiple EUIs found in arguments, considering the first")
+				joinEUIHex = args[0]
+				startDevEUIHex = args[1]
+			}
+			if joinEUIHex == "" {
+				return errNoEndDeviceTemplateJoinEUI
+			}
+			if startDevEUIHex == "" {
+				return errNoEndDeviceTemplateStartDevEUI
+			}
+
+			var joinEUI types.EUI64
+			if err := joinEUI.UnmarshalText([]byte(joinEUIHex)); err != nil {
+				return err
+			}
+			var startDevEUI types.EUI64
+			if err := startDevEUI.UnmarshalText([]byte(startDevEUIHex)); err != nil {
+				return err
+			}
+			devEUIInt := binary.BigEndian.Uint64(startDevEUI[:])
+
+			count, _ := cmd.Flags().GetInt("count")
+			for {
+				var template ttnpb.EndDeviceTemplate
+				if _, err := inputDecoder.Decode(&template); err != nil {
+					if err == stdio.EOF {
+						return nil
+					}
+					return err
+				}
+
+				for i := 0; i < count; i++ {
+					res := template
+
+					var devEUI types.EUI64
+					binary.BigEndian.PutUint64(devEUI[:], devEUIInt)
+					devEUIInt++
+
+					res.EndDevice.DeviceID = fmt.Sprintf("eui-%s", strings.ToLower(devEUI.String()))
+					res.EndDevice.JoinEUI = &joinEUI
+					res.EndDevice.DevEUI = &devEUI
+					res.FieldMask.Paths = ttnpb.BottomLevelFields(append(res.FieldMask.Paths,
+						"ids.device_id",
+						"ids.join_eui",
+						"ids.dev_eui",
+					))
+
+					if err := io.Write(os.Stdout, config.OutputFormat, &res); err != nil {
+						return err
+					}
+				}
+			}
+		},
+	}
+	endDeviceTemplatesListFormats = &cobra.Command{
+		Use:               "list-formats",
+		Short:             "List available end device template formats",
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dtc, err := api.Dial(ctx, config.DeviceTemplateConverterGRPCAddress)
+			if err != nil {
+				return err
+			}
+			res, err := ttnpb.NewEndDeviceTemplateConverterClient(dtc).ListFormats(ctx, ttnpb.Empty)
+			if err != nil {
+				return err
+			}
+			return io.Write(os.Stdout, config.OutputFormat, res)
+		},
+	}
+	endDeviceTemplatesConvertCommand = &cobra.Command{
+		Use:               "convert [format-id]",
```

Comment:
`ttn-lw-cli end-device template convert` sound like we're converting a template to something. Perhaps rename this to something like `from-data`?